gammapy/obs/observation.py

```diff
@@ -319,3 +339,569 @@ def select_observations(self, selection=None):
+        """List of groups (`~numpy.ndarray`)"""
+        return self.obs_groups_table['GROUP_ID'].data
+
+    def axes_to_table(self, axes):
+        """Fill the observation group axes into a table.
+
+        Define one row for each possible combination of the
+        observation group axis bins. Each row will represent
+        an observation group.
+
+        Parameters
+        ----------
+        axes : `~gammapy.obs.ObservationGroupAxis`
+            List of observation group axes.
+
+        Returns
+        -------
+        table : `~astropy.table.Table`
+            Table containing the observation group definitions.
+        """
+        # define table column data
+        column_data_min = []
+        column_data_max = []
+        # loop over observation axes
+        for i_axis in np.arange(len(axes)):
+            if axes[i_axis].format == 'bin_values':
+                column_data_min.append(axes[i_axis].bins)
+                column_data_max.append(axes[i_axis].bins)
+            elif axes[i_axis].format == 'bin_edges':
+                column_data_min.append(axes[i_axis].bins[:-1])
+                column_data_max.append(axes[i_axis].bins[1:])
+
+        # define grids of column data
+        ndim = len(axes)
+        s0 = (1,)*ndim
+        expanding_arrays = [x.reshape(s0[:i] + (-1,) + s0[i + 1::])
+                            for i, x in enumerate(column_data_min)]
+        column_data_expanded_min = np.broadcast_arrays(*expanding_arrays)
+        expanding_arrays = [x.reshape(s0[:i] + (-1,) + s0[i + 1::])
+                            for i, x in enumerate(column_data_max)]
+        column_data_expanded_max = np.broadcast_arrays(*expanding_arrays)
+
+        # recover units
+        for i_dim in np.arange(ndim):
+            column_data_expanded_min[i_dim] = recover_units(column_data_expanded_min[i_dim],
+                                                            column_data_min[i_dim])
+            column_data_expanded_max[i_dim] = recover_units(column_data_expanded_max[i_dim],
+                                                            column_data_max[i_dim])
+
+        # define table columns
+        columns = []
+        for i_axis in np.arange(len(axes)):
+            if axes[i_axis].format == 'bin_values':
+                columns.append(Column(data=column_data_expanded_min[i_axis].flatten(),
+                                      name=axes[i_axis].name))
+            elif axes[i_axis].format == 'bin_edges':
+                columns.append(Column(data=column_data_expanded_min[i_axis].flatten(),
+                                      name=axes[i_axis].name + "_MIN"))
+                columns.append(Column(data=column_data_expanded_max[i_axis].flatten(),
+                                      name=axes[i_axis].name + "_MAX"))
+
+        # fill table
+        table = Table()
+        for i, col in enumerate(columns):
+            table.add_column(col)
+
+        return table
+
+    def table_to_axes(self, table):
```

Comment:
Make this a `staticmethod`, to make it clear that `self` isn't used / modified.

```python
@staticmethod
def table_to_axes(table):
```
(and then change the call sites to call `ObservationGroups.table_to_axes`)